README.rst

```diff
@@ -475,41 +475,91 @@ any operations on iRODS client APIs.  This may also be overridden, on a per-thre
 Rule Execution
 --------------
 
-A simple example for executing an iRODS rule from the Python client is as follows.  Suppose we have a rule file
+A simple example of how to execute an iRODS rule from the Python client is as follows.  Suppose we have a rule file
 :code:`native1.r` which contains a rule in native iRODS Rule Language::
 
   main() {
-      writeLine("*stream", 
+      writeLine("*stream",
                 *X ++ " squared is " ++ str(double(*X)^2) )
   }
 
   INPUT *X="3", *stream="serverLog"
   OUTPUT null
 
-The following Python client code will run that rule and produce output in the
+The following Python client code will run the rule and produce the appropriate output in the
 irods server log::
 
-  r = irods.rule.Rule( ses, rule_file = 'native1.r')
-  r.execute( )
+  r = irods.rule.Rule( session, rule_file = 'native1.r')
+  r.execute()
 
-Release v1.1.0 improvements include the ability to target a specific rule engine instance by name, so if other rule engine instances are present, we may want
-to execute using::
+Release v1.1.0 includes the ability to target a specific rule engine instance by name, so if other rule engine instances are present, we may want
+to instantiate using::
 
-  r.execute( instance_name = 'irods_rule_engine_plugin-irods_rule_language-instance')
+  Rule( ... , instance_name = 'irods_rule_engine_plugin-irods_rule_language-instance' )
 
-Incidentally, if we wanted to inform the native1.r rule code to print stdout instead, we could also set parameters for the rule and change the OUTPUT parameter
+Additionally, if we wanted to have the :code:`native1.r` rule code print to stdout instead, we could also set parameters for the rule and change the OUTPUT parameter
 to accommodate the output stream::
 
-  r = irods.rule.Rule( ses, rule_file = 'native1.r',
+  r = irods.rule.Rule( session, rule_file = 'native1.r',
              instance_name = 'irods_rule_engine_plugin-irods_rule_language-instance',
              params={'*stream':'"stdout"'} , output = 'ruleExecOut' )
   output = r.execute( )
   if output and len(output.MsParam_PI):
       buf = output.MsParam_PI[0].inOutStruct.stdoutBuf.buf
       if buf: print(buf.rstrip(b'\0').decode('utf8'))
 
-Changing  the integer input to the squaring operation in this example is left as an exercise for the reader :) .
+(Changing the input value to be squared in this example is left as an exercise for the reader!)
 
+To deal with errors resulting from rule execution failure, two approaches can be taken. Suppose we
+have define this in the :code:`/etc/irods/core.re` rule-base::
```

Comment:
not quite good grammar... "have define this"

Reply:
ok thanks, will correct!